internal/pkg/cli/task_run.go

```diff
@@ -575,6 +583,54 @@ func (o *runTaskOpts) runTaskCommandFromECSService(sess *session.Session, cluste
 	return cmd, nil
 }
 
+func (o *runTaskOpts) runTaskCommandFromWorkload(sess *session.Session, appName, envName, workloadName string) (cliStringer, error) {
+	workloadType, err := o.workloadType(appName, workloadName)
+	if err != nil {
+		return nil, err
+	}
+
+	var cmd cliStringer
+	switch workloadType {
+	case workloadTypeJob:
+		cmd, err = o.runTaskRequestFromJob(ecs.New(sess), appName, envName, workloadName)
+		if err != nil {
+			return nil, fmt.Errorf("generate task run command from job %s of application %s deployed in environment %s: %w", workloadName, appName, envName, err)
+		}
+	case workloadTypeSvc:
+		cmd, err = o.runTaskRequestFromService(ecs.New(sess), appName, envName, workloadName)
+		if err != nil {
+			return nil, fmt.Errorf("generate task run command from service %s of application %s deployed in environment %s: %w", workloadName, appName, envName, err)
+		}
+	default:
+		return nil, fmt.Errorf("workload %s is neither a service nor a job", workloadName)
+	}
+	return cmd, nil
+}
+
+func (o *runTaskOpts) workloadType(appName, workloadName string) (string, error) {
+	_, err := o.store.GetJob(appName, workloadName)
```

Comment:
Hmmm. What if a job and a service have the same name? If the service is intended, it will be determined to be a job here, no?

Reply:
+1; good callout

Reply:
I think it's impossible to have a service and a job deployed with the same name. They would have encountered issues during deployment because they'd have tried to create to ssm parameter store with the same name. (The ssm parameter for both service and job have the format `/copilot/applications/app/components/workload`

Reply:
Assuming by the time users run `task run --generate-cmd`, they already have the service or the job deployed - they wouldn't have encountered this name conflict.

Reply:
Oh, you're right-- I only tried to init, didn't try to deploy. Yeah, the latter manifest doesn't get created because it "already exists." Thanks!